ui/analyze_window.cpp

```diff
@@ -445,108 +433,73 @@ void AnalyzeDialog::OnDeviceListRefresh()
 }
 
 //--------------------------------------------------------------------------------------------------
-void AnalyzeDialog::OnOpenFile()
+void AnalyzeDialog::OnAnalyzeCaptureStarted(const QString &file_path)
 {
-    QString supported_files = QStringLiteral("GFXR files (*.gfxr)");
-    QString file_name = QFileDialog::getOpenFileName(this,
-                                                     "Open Capture File",
-                                                     Settings::Get()->ReadLastFilePath(),
-                                                     supported_files);
+    // Clear members for previous session
+    OnAnalyzeCaptureEnded();
 
-    if (!file_name.isEmpty())
+    // Validate capture
+    if (file_path.isEmpty())
     {
-        // Convert the filename to a string to perform a replacement.
-        std::string potential_asset_name(file_name.toStdString());
-
-        const std::string trim_str = "_trim_trigger";
-        const std::string asset_str = "_asset_file";
-
-        // Find and replace the last occurrence of "trim_trigger" part of the filename.
-        size_t pos = potential_asset_name.rfind(trim_str);
-        if (pos != std::string::npos)
-        {
-            potential_asset_name.replace(pos, trim_str.length(), asset_str);
-        }
-
-        // Create a path object to the asset file.
-        std::filesystem::path asset_file_path(potential_asset_name);
-        asset_file_path.replace_extension(".gfxa");
-
-        // Check if the required asset file exists.
-        bool asset_file_exists = std::filesystem::exists(asset_file_path);
-
-        if (!asset_file_exists)
+        qDebug() << "OnAnalyzeCaptureStarted(): empty filename";
+        return;
+    }
+    std::filesystem::path    local_gfxr_parse = file_path.toStdString();
+    Dive::ComponentFilePaths component_paths = {};
```

Comment:
if (auto ret = Dive::GetComponentFilesHostPaths(local_gfxr_parse.parent_path(),
                                               local_gfxr_parse.stem().string()); ret.ok())
{
    component_paths = *ret;
} else {
    // Handle the error (ret.status())
}

Reply:
Oh, thanks for bring this up. I completely missed that comment. I feel that this way uses the C++ language feature specifically designed for initialization and checking (if with initializer, available since C++17). It also keeps the logic for initializing, checking, and extracting the value together in one line, improving local coherence. 

But after thinking about it, i think it is just a personal preference and both are fine as long as we can keep a consistent style in the code, and i am ok if you prefer to keep it this way. (Still interesting to have this discussion :) )